pandas/core/arrays/masked.py

```diff
@@ -40,6 +40,17 @@ class BaseMaskedDtype(ExtensionDtype):
     def numpy_dtype(self) -> np.dtype:
         raise AbstractMethodError
 
+    @classmethod
+    def construct_array_type(cls) -> Type["BaseMaskedArray"]:
+        """
+        Return the array type associated with this dtype.
+
+        Returns
+        -------
+        type
+        """
+        raise NotImplementedError
```

Comment:
should use AbstractMethodError

Reply:
We actually use NotImplementedError in the base dtype class for this, so this is consistent with that (not fully sure why we do that though, there might be a reason)

Reply:
kk yeah we should actually change that as this is a better error message (but ok)